pipelinewise/cli/pipelinewise.py

```diff
@@ -1729,6 +1731,42 @@ def _run_post_import_tap_checks(
 
         return errors
 
+    def _cleanup_tap_state_file(self) -> None:
+        tables = self.args.tables
+        state_file = self.tap['files']['state']
+        if tables:
+            self._clean_tables_from_bookmarks_in_state_file(state_file, tables)
+        else:
+            try:
+                os.remove(state_file)
+            except FileNotFoundError:
+                pass
+
+    @staticmethod
+    def _clean_tables_from_bookmarks_in_state_file(state_file_to_clean: str, tables: str) -> None:
+        try:
+            with open(state_file_to_clean, 'r+', encoding='UTF-8') as state_file:
+                state_data = json.load(state_file)
+                bookmarks = state_data.get('bookmarks')
+                list_of_tables = tables.split(',')
+                if bookmarks:
+                    cleanedup_bookmarks = {
+                        bookmark_name: bookmarks[bookmark_name]
+                        for bookmark_name in bookmarks
+                        if not any(table_name in bookmark_name for table_name in list_of_tables)
+                    }
```

Comment:
This could be written in a more efficient & simpler way:
```suggestion
               list_of_tables = tables.split(',')
                if bookmarks:
                    for table in list_of_tables:
                         bookmarks.pop(table_name, None)
```

Reply:
I think your suggestion for line #1753 should be like this : `for table_name in list_of_tables:`
Any how, If you run `test_command_sync_tables_cleanup_state_if_file_exists_and_table_argument` after this suggestion, you will see it fails.
The reason: your simpler way only removes the bookmark if `table_name` is exactly same as bookmark key, but in reality it is not! the bookmark keys have some prefixes same as what is implemented in the tests.
i.e: bookmark in state file : `taped-table_name` but tables argument: `table_name`

Reply:
Are you sure about that? the prefix in the bookmark keys is the DB/schema name, not the tap id, and when you run the `sync_table` command, you have to provide those as well, not just `table1,table2...`

So in this failing test case, the input is incorrect.

Reply:
I checked again the tap codes, you are right. I will fix the test and implementation.

Reply:
- [ ] Make the code simpler!